cloudevents/extensions/authcontext.md

```diff
@@ -0,0 +1,55 @@
+# Auth Context
+
+This extension embeds information about the principal which triggered an occurence. This allows consumers of the
+CloudEvent to perform user-dependent actions without requiring the user ID to
+be embedded in the `data` or `source` field.
+
+This extension is purely informational and is not intended to secure CloudEvents.
+
+## Notational Conventions
+
+As with the main [CloudEvents specification](../spec.md), the key words "MUST",
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+However, the scope of these key words is limited to when this extension is
+used. For example, an attribute being marked as "REQUIRED" does not mean
+it needs to be in all CloudEvents, rather it needs to be included only when 
+this extension is being used.
+
+## Attributes
+
+### authtype
+
+- Type: `String`
+- Description: An enum representing the type of principal that triggered the occurence.
+Valid values are:
+  - `app_user`: An end user of an application. Examples include an AWS cognito,
+    Google Cloud Identity Platform, or Azure Active Directory user.
+  - `user`: A user account registered in the infrastructure. Examples include
+    developer accounts secured by IAM in AWS, Google Cloud Platform, or Azure.
+  - `service_account`: A non-user principal used to identify a service.
+  - `api_key`: A non-user API key
+  - `system`: An obscured identity used when a cloud platform or other system
+    service triggers an event. Examples include a database record which
+    was deleted based on a TTL.
+- Constriants
+  - REQUIRED
+  - SHOULD be one of the above enum values
+  - If the enum values are not sufficient for a Producer, they SHOULD amend this spec to include the new enum value.
+
+### authid
+- Type: `String`
+- Description: A unique identifier of the principal that triggered the occurence. This might, for example, be a unique ID in an identity database (userID), an email of a platform user or service account, or the label for an API key.
+- Constraints
+  - OPTIONAL
```

Comment:
Should this be REQUIRED? Seems like w/o this info the extension doesn't provide too much value to the consumer. Unless there are cases in just knowing "some" app_user caused it???

Reply:
I think many clouds will not have an ID for api keys and most clouds will not want to have an ID for system changes.

Reply:
ah ok - thanks!

Reply:
Also I just added "unauthenticated" to the enum for "authtype" so that a provider can indicate that this extension is supported but the request had no authentication. That would obviously not come with an authid either.

Reply:
ok  can you sign your commits?